src/operations/StandardOperation.cpp

```diff
@@ -558,4 +558,207 @@ void StandardOperation::invert() {
                        " is not supported.");
   }
 }
+void StandardOperation::dumpOpenQASM3(std::ostream& of,
```

Comment:
Same as for the non-unitary operation goes here: any chance this can be consolidated with the OQ2 method? For example, by extracting common parts?